src/HomeKit/HMHome.cs

```diff
@@ -80,6 +82,16 @@ public partial class HMHome
 			return GetServices (arr.ToArray ());
 		}
 
+		class ServiceTypeList<T> : List<T> {
```

Comment:
This saves a ton of null checks above from lines 17 - 80, but this may be hiding the null check. Opinions?